src/librustdoc/formats/item_type.rs

```diff
@@ -95,6 +95,10 @@ impl<'a> From<&'a clean::Item> for ItemType {
             clean::ForeignFunctionItem(..) => ItemType::Function, // no ForeignFunction
             clean::ForeignStaticItem(..) => ItemType::Static,     // no ForeignStatic
             clean::MacroItem(..) => ItemType::Macro,
+            // Is this a good idea?
+            clean::AttrMacroItem => ItemType::ProcAttribute,
+            // Is this a good idea?
+            clean::DeriveMacroItem => ItemType::ProcDerive,
```

Comment:
It seems fairly confusing to me, I don't see the advantage (is it so `item_module` groups things correctly? if that's the case, then the first commit on its own would not be correct, since it would categorize these macros as `ItemType::Macro`), and it already causes a bit of weirdness in `html_filename` which would not be needed if these were categorized as `Macro` instead.

if we did want to make this change, I would at least want to change the names of the item types to not imply they must be derives.